app/UI/relay-recommend-list.tsx

```diff
@@ -9,8 +9,40 @@ import {
     HoverButtonBackgroudColor,
     PrimaryTextColor,
 } from "./style/colors.ts";
+import { emitFunc } from "../event-bus.ts";
+import { RelayConfigChange } from "./setting.tsx";
+import { RelayConfig } from "./relay-config.ts";
+
+type RelayRecommendListProps = {
+    relayConfig: RelayConfig;
+    emit: emitFunc<RelayConfigChange>;
+};
+
+type RelayRecommendListState = {
+    relays: string[];
+};
+
+export class RelayRecommendList extends Component<RelayRecommendListProps, RelayRecommendListState> {
+    state: Readonly<RelayRecommendListState> = {
+        relays: this.computeRecommendedRelays(),
+    };
+
+    computeRecommendedRelays() {
+        // remove the relays that are already in the relayConfig
+        return recommendedRelays.filter((r) => !this.props.relayConfig.getRelayURLs().has(r));
```

Comment:
This is not a state because it can be determined at the init time of the component. State is things that changes during the life time of a component instance.